ColorSwitcher/Model/GIFManager.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  GIFManager.swift
+//  ColorSwitcher
+//
+//  Created by Matt M Smith on 9/28/20.
+//
+
+import Foundation
+
+protocol GIFManagerDelegate {
+    func getGIFModel(title: String, urlString: String)
+}
+
+struct GIFManager {
+    var delegate: GIFManagerDelegate?
+    
+    private let apiKey = "0OZKMCWSOD4M"
+    private let baseURL = "https://api.tenor.com/v1/search?"
+    
+    func fetchGIF(from query: String) {
+        let formattedQuery = query.replacingOccurrences(of: " ", with: "+")
+        let urlString = "\(baseURL)?api_key=\(apiKey)&q=\(formattedQuery)&contentfilter=high&media_filter=minimal&limit=1"
+        
+        if let url = URL(string: urlString) {
+            let session = URLSession(configuration: .default)
+            let task = session.dataTask(with: url) { (data, _, error) in
+                if let error = error {
+                    print(error)
+                }
+                if let data = data {
+                    let model = self.parse(jsonData: data)
+
+                    self.delegate?.getGIFModel(title: model.title, urlString: model.gifURLString)
+                }
+            }
+            task.resume()
+        }
+    }
+    
+    private func parse(jsonData data: Data) -> GIFModel {
+        let decoder = JSONDecoder()
+        var title = ""
+        var gifURLString = ""
+        
+        do {
+            let decodedData = try decoder.decode(GIFData.self, from: data)
+            title = decodedData.results[0].title
```

Comment:
If you want to get Swifty you could say `decodedData.results.first.title`.

Reply:
But the old timers like `[0]`.

Reply:
But that brings up an interesting point. `first` returns an Optional String and rightly so. There might not be a String at index 0. So you're playing with run time fire accessing by index 0. I mean, you basically know it's going to be there but for safety you might consider using `first` to acknowledge it might not exist and unwrap the Optional.

Reply:
I'm in the process of making the change and realize this is a little more than just unwrapping an optional. I'll have to make sure that parse(:) will also need to return an optional as well. 

I'm curious about one thing though. When I'm instantiating the decoder object, isn't that when I'm confirming that I will indeed have something at index 0? If there isn't then it should throw an error, and drop me down to the catch block, right?

Reply:
At the point where `decoder.decode(GIFData.self, from: data) completes successfully you're guaranteed to have a valid GIFData object. The tricky part is that while the results array is guaranteed to exist, it might be empty. Like you could get {"results":[]}, which would decode successfully. If you have one or more objects in `results`, each one will have a `media` array that exists but could be empty. Does that answer what you were looking for?

Reply:
Yes! Thank you! That's exactly what I was looking for.